### ISSUE TITLE
Fix false positives for parametric mixins in selector-class-pattern

### ISSUE THREAD

**Issue by SevenOutman** [stylelint/stylelint#5258]
> Clearly describe the bug

Getting `selector-class-pattern` violations on LESS mixin definitions.

> Which rule, if any, is the bug related to?

`selector-class-pattern`

> What code is needed to reproduce the bug?

```less
.my-mixin(@x: 5) {
  line-height: @x;
}
```

> What stylelint configuration is needed to reproduce the bug?
```js
{
  "rules": {
    // kebab-case
    'selector-class-pattern': '^[a-z0-9]+(-?[a-z0-9]+)*$'
  }
}
```

> Which version of stylelint are you using?

`13.12.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "src/**/*.less" --syntax less`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to [LESS parametric mixins](https://lesscss.org/features/#mixins-feature-mixins-parametric-feature).

> What did you expect to happen?

The `selector-class-pattern` rule ignores LESS mixins [as documented](https://stylelint.io/user-guide/rules/selector-class-pattern)

> What actually happened (e.g. what warnings or errors did you get)?

Styelint seems not to recognize it as a mixin.

```shell
Expected class selector ".my-mixin(@x: 5)" to      selector-class-pattern
match pattern "^[a-z0-9]+(-?[a-z0-9]+)*$"
```

**Comment by jeddy3** [comment]
@SevenOutman Thanks for the report and clear write-up.

Yes, it appears to be a bug. The heuristics in the [`isStandardSyntaxRule` util](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxRule.js) need to be tweaked to catch this.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by smile33** [comment]
Hope to fix it soon

**Comment by jeddy3** [comment]
@smile33 Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by mattxwang** [comment]
Looks like this should be closed by #5378, but feel free to re-open if needed!

**Comment by ybiquitous** [comment]
@mattxwang Good catch! Thank you 😄
